### ISSUE TITLE
Can't copy/paste code from the documentation

### ISSUE THREAD

**Issue by kpym** [loopspace/hobby#3]
When I copy/paste some code from the pdf of the documentation a lot of white spaces are added and the code is not working as is. 
Here is an example of copy/paste  using Chrome:
```
\ be gin { t i k z p i c t u r e } [ s c a l e =. 5]
\draw [ p o s t a c ti o n=show cu r ve c o n t r o l s ]
( 0 , 0 ) t o [ cl o s e d , cu r ve through = { ( 1 ,. 5 ) . . ( 2 , 0 ) . .
( 3 , . 5 ) } ] ( 4 , 0 ) ;
\draw [ x s h i f t =5cm, u se Hobby s h o r t c u t , p o s t a c ti o n=show
cu r ve c o n t r o l s ]
( [ c l o s e d ] 0 , 0 ) . . ( 1 , . 5 ) . . ( 2 , 0 ) . . ( 3 , . 5 ) . . ( 4 , 0 ) ;
\end{ t i k z p i c t u r e }
```
Here is an example of copy/paste  using Sumatra (Windows):
```
\begin{ tikzpicture }[ scale =.5]
\draw [ postaction=show curve controls ]
(0 ,0) to [ closed , curve through ={(1 ,.5) . . (2 ,0) . .
(3 ,.5) }] (4 ,0) ;
\draw [ xshift=5cm, use Hobby shortcut , postaction=show
curve controls ]
([ closed ]0 ,0) . . (1 ,.5) . . (2 ,0) . . (3 ,.5) . . (4 ,0) ;
\end{ tikzpicture }
```
The Sumatra paste is better but the dots `. .` have space between. And in some other codes the minus sign is not the good one.

And in the documentation the only bold keyword in the code is `end` . 

Probably for all this reasons you can use another listing package that produce nicer and more copy/paste friendly output ;)

**Comment by loopspace** [comment]
Must admit, I don't often cut-and-paste from PDFs so I've little experience in this matter.  If you know of a better package for verbatim code, please do let me know.

In the meantime, the source of the documentation is available here so you can cut-and-paste from that.

**Comment by kpym** [comment]
I would like to help, but when I try to compile `hobby_doc.tex` I obtain an error  

```
... [11] (example.out) (example.out
! Undefined control sequence.
<argument> \g__prg_map_int

l.2 ...obby path={saved}] (0,0) .. (1,1) .. (2,0);
```
This error seems to be generated by `save Hobby path={saved}`  on line 396 of the documentation. Is one of my packages not up to date ? (my installation is pretty recent)

**Comment by kpym** [comment]
I think I have found where the copy/paste problem comes from : you do not use a monospaced font and `listings` package compensate this by inserting spaces. To correct this you can say : `basicstyle=\small\ttfamily` (to use monospaced font) and `keepspaces=true,columns=fullflexible` (to not add additional spaces or drop them in any case).

So I recommend to use the style : 
```latex
\lstset{
  breakatwhitespace=true,
  breaklines=true,
  language=[LaTeX]TeX,
  basicstyle=\small\ttfamily,
  keepspaces=true,
  columns=fullflexible
}
```

**Comment by kpym** [comment]
Out of topic. 

If you want to color your keywords you can add : 
```
\lstset{
  emphstyle={\color{red}},
  emph={
    angle,
    closed,
    controls,
    curl,
    curve,
    designated,
    finish,
    Hobby,
    hobby,
    in,
    next,
    out,
    path,
    quick,
    shortcut,
    show,
    tension,
    through,
    use
  } 
}
```
Which produce the following output : 
![](https://i.imgur.com/33DxOti.png)

And if you want to color also the tikz words you can add : 
```
\lstset{
  emphstyle={[2]\color{blue!70!black}},
  emph={[2]
    addplot,
    axis,
    coordinates,
    draw,
    grid,
    help,
    lines,
    plot,
    postaction,
    scale
    smooth,
    tikzpicture,
    to,
    xshift,
  }
}
```

![](https://i.imgur.com/rHO43gc.png)

**Comment by loopspace** [comment]
These all look great suggestions.  I've updated the documentation `.tex` file accordingly.

I also think I've fixed the issue that was preventing you compiling the documentation.

If you wouldn't mind trying to compile the documentation and seeing if you can cut-and-paste from it then this issue can be closed.

Thanks!

**Comment by kpym** [comment]
You should remove [line 81](https://github.com/loopspace/hobby/blob/b6cf4d29a27406948fdf95a46281ac2aaca25d9c/hobby_doc.tex#L81) I think. This line overwrite `basicstyle` and `language`, and is useless (or not ?). I can now compile the doc without error ;)

**Comment by kpym** [comment]
This has been solved by #4 and 80fabef.
